core/drivers/crypto/caam/acipher/caam_ecc.c

```diff
@@ -0,0 +1,724 @@
+				       const struct ecc_public_key *inkey,
+				       size_t size_sec)
+{
+	enum caam_status retstatus = CAAM_OUT_MEMORY;
+	size_t x_size = 0;
+	size_t y_size = 0;
+
+	ECC_TRACE("ECC Convert Public Key size %zu bytes", size_sec);
+
+	/* Point (x y) is twice security key size */
+	retstatus = caam_calloc_buf(&outkey->xy, 2 * size_sec);
+	if (retstatus != CAAM_NO_ERROR)
+		return retstatus;
+
+	/* Copy x and y and get the number of bytes to pad with 0's */
+	x_size = crypto_bignum_num_bytes(inkey->x);
+	crypto_bignum_bn2bin(inkey->x, outkey->xy.data + size_sec - x_size);
+
+	y_size = crypto_bignum_num_bytes(inkey->y);
+	crypto_bignum_bn2bin(inkey->y, outkey->xy.data + 2 * size_sec - y_size);
+
+	cache_operation(TEE_CACHECLEAN, outkey->xy.data, outkey->xy.length);
+
+	return CAAM_NO_ERROR;
+}
+
+/*
+ * Convert Crypto ECC Key to local ECC Keypair Key
+ * Don't convert the exponent e not used in decrytion
+ *
+ * @outkey    [out] Output keypair in local format
+ * @inkey     Input key in TEE Crypto format
+ * @size_sec  Security size in bytes
+ */
+static enum caam_status do_keypair_conv(struct caam_ecc_keypair *outkey,
+					const struct ecc_keypair *inkey,
+					size_t size_sec)
+{
+	enum caam_status retstatus = CAAM_OUT_MEMORY;
+	size_t d_size = 0;
+
+	ECC_TRACE("ECC Convert Keypair size %zu bytes", size_sec);
+
+	/* Private key is only scalar d of sec_size bytes */
+	retstatus = caam_calloc_buf(&outkey->d, size_sec);
+	if (retstatus != CAAM_NO_ERROR)
+		return retstatus;
+
+	/* Get the number of bytes of d to pad with 0's */
+	d_size = crypto_bignum_num_bytes(inkey->d);
+	crypto_bignum_bn2bin(inkey->d, outkey->d.data + size_sec - d_size);
+
+	cache_operation(TEE_CACHECLEAN, outkey->d.data, outkey->d.length);
+
+	return CAAM_NO_ERROR;
+}
+
+/*
+ * Convert TEE ECC Curve to CAAM ECC Curve
+ *
+ * @tee_curve  TEE ECC Curve
+ */
+static enum caam_ecc_curve get_caam_curve(uint32_t tee_curve)
+{
+	enum caam_ecc_curve caam_curve = CAAM_ECC_UNKNOWN;
+
+	if (tee_curve > 0 &&
+	    tee_curve < CAAM_ECC_MAX + TEE_ECC_CURVE_NIST_P192) {
+		/*
+		 * Realign TEE Curve assuming NIST_P192 is the first entry in
+		 * the list of supported ECC curves.
+		 */
+		caam_curve = tee_curve - TEE_ECC_CURVE_NIST_P192
+			     + CAAM_ECC_P192;
+	}
+
+	return caam_curve;
+}
+
+/*
+ * Allocate a ECC keypair
+ *
+ * @key        Keypair
+ * @size_bits  Key size in bits
+ */
+static TEE_Result do_allocate_keypair(struct ecc_keypair *key, size_t size_bits)
+{
+	ECC_TRACE("Allocate Keypair of %zu bits", size_bits);
+
+	/* Initialize the key fields to NULL */
+	memset(key, 0, sizeof(*key));
+
+	/* Allocate Secure Scalar */
+	key->d = crypto_bignum_allocate(size_bits);
+	if (!key->d)
+		goto out;
+
+	/* Allocate Public coordinate X */
+	key->x = crypto_bignum_allocate(size_bits);
+	if (!key->x)
+		goto out;
+
+	/* Allocate Public coordinate Y */
+	key->y = crypto_bignum_allocate(size_bits);
+	if (!key->y)
+		goto out;
+
+	return TEE_SUCCESS;
+
+out:
```

Comment:
For this usage we normally name this label `err:`

Reply:
I will replace ```out```  with ```err``` for this function and ```do_allocate_publickey()```